initial commit

AlexLanzano opened the pull request.

Commit:
Migrated old repo to new repo

cjp256 commented on include/exit_handler/xen_structs.h:
```diff
@@ -0,0 +1,100 @@
+
+}shared_info_t;
+
+shared_info_t *shared_info;
```
I don't think this belongs here.

cjp256 commented on src/xen_exit_handler/src/xen_exit_handler.cpp:
```diff
@@ -0,0 +1,266 @@
+{
+    bfdebug << "console io" << bfendl;
+
+    switch(rdi) {
```
formatting is not consistent throughout this module (indent, bracing, etc)

Commit:
fixed issues from repo rename

cjp256 commented on test_drivers/console_io/console_io.c:
```diff
@@ -0,0 +1,64 @@
+
+    /* Check if hypercall_page was populated by Bareflank */
+
+    if ((char *)hypercall_page == 0) {
```
supposed to be *(uint64_t *)hypercall_page == 0 ?

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+    struct shared_info *shared_info;
+} thread_args_t;
+
+struct shared_info *shared_info;
```
globals that are really module-scope, should be declared as static

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+                  : "memory"
+                  );
+}
+inline void make_hypercall3(unsigned long rax, unsigned long rdi,
```
i assume these are xen-ABI hypercalls? maybe indicate as such?

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+
+    make_hypercall2(INIT_SHARED_INFO, (unsigned long)shared_info, tsc_khz);
+    return true;
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+
+
+
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+
+static int __init driver_start(void)
+{
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+
+    return true;
+}
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+        return false;
+    }
+    return true;
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+    do_exit(0);
+    return 0;
+}
+
```
nit: check your whitespacing

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+
+
+/**
+ * clocks_calc_mult_shift - calculate mult/shift factors for scaled math of clocks
```
credit where this was copied from?

cjp256 commented on test_drivers/console_io/console_io.c:
```diff
@@ -0,0 +1,64 @@
+{
+    uint32_t base, eax, signature[3];
+
+    for (base = 0x40000000; base < 0x40010000; base += 0x100) {
```
credit where copied from

cjp256 commented on test_drivers/pvclock/pvclock.c:
```diff
@@ -0,0 +1,286 @@
+MODULE_LICENSE("GPL");
+
+#define NANOSECONDS(delta) ((1000000ULL * delta) / tsc_khz)
+#define INIT_SHARED_INFO 100
```
these are duplicated from namespace test_hypercall (maybe use a common header?)

cjp256 commented on src/xen_exit_handler/src/xen_exit_handler.cpp:
```diff
@@ -0,0 +1,266 @@
+
+    //shared_info->vcpu_info[0].time.tsc_timestamp = new_tsc;
+    //shared_info->vcpu_info[0].time.system_time = system_time;
+    shared_info->wc.sec = seconds;
```
split out the update_shared_info()

cjp256 replied:
i think during this, xen does something like setting the lowest bit during the update

cjp256 replied:
(in the version field)

Commit:
Cleaned up

Commit:
Got rid of update_fake_clock.

Time references for the hypervisor are only updated once when the
driver is loaded.
